### ISSUE TITLE
FTC Driver Station stuck on "Please wait while we update the Wifi Direct settings on this device"

### ISSUE THREAD

**Issue by higgs241** [SwerveRobotics/ftc_app#73]
Installed the app on the robot controller phone and side loaded the driver station app, but now the driver station app is stuck on that screen.

**Comment by cmacfarl** [comment]
We think this is because the DS misses a "done" event for wifi setup but otherwise the initialization is complete.  I'll wait a couple minutes and then just hit back and everything seems to be fine.

**Comment by higgs241** [comment]
solution is to reset the wifi direct settings in the robot controller app

**Comment by higgs241** [comment]
Never mind I'm still having this issue.

**Comment by lukethecoder64** [comment]
Can confirm. Usually happens to us when changing to a wifi direct channel
that has other teams on it.

On Monday, February 22, 2016, higgs241 <notifications@github.com> wrote:

> Never mind I'm still having this issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SwerveRobotics/ftc_app/issues/73#issuecomment-187425420>
> .
>


-- 
Luke
     __
    /   /
   /   /
  /   /
 /   /______
/_________/

**Comment by nitrocalcite** [comment]
We have this issue sometimes.  It's solved by repeatedly pressing the back button until to get to the app.  No clue why, it's really odd.

**Comment by cmacfarl** [comment]
Should be fixed in the next release of the SDK.

**Comment by Fromen8923** [comment]
I found a cause for this, and a workaround. If you have both the RC and DS on one phone is when I've seen it happen: when the DS is started for the first time, it configures the WiFi Direct settings, and finishes properly. Then opening the RC does the same thing: it configures properly. Opening the DS after the RC is when it gets stuck on the configuration screen. So it seems to me that when the RC does its configuration, it causes some issue that doesn't allow the DS to configure properly.

The workaround I found is to simply restart the phone. My guess is that resets all of the WiFi Direct settings, meaning the DS can configure it properly.
